components/openweather_api/openweather_api.app.mjs

```diff
@@ -1,11 +1,97 @@
+import { axios } from "@pipedream/platform";
+import constants from "./common/constants.mjs";
+
 export default {
   type: "app",
   app: "openweather_api",
-  propDefinitions: {},
+  propDefinitions: {
+    appId: {
+      type: "string",
+      label: "App Id",
+      description: "Please refer to [Doc here](https://home.openweathermap.org/api_keys) for your app ID.",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _authToken() {
+      return this.$auth.oauth_access_token;
+    },
+    getHeader() {
+      return {
+        "Content-Type": "application/json",
+      };
+    },
+    getLocationAPI() {
+      const {
+        GEO_PATH,
+        GEO_VERSION_PATH,
+      } = constants;
+      return GEO_PATH + GEO_VERSION_PATH;
+    },
+    getWeatherAPI() {
+      const {
+        WEATHER_PATH,
+        WEATHER_VERSION_PATH,
+      } = constants;
+      return WEATHER_PATH + WEATHER_VERSION_PATH;
+    },
+    getForcastAPI() {
+      const {
+        FORCAST_PATH,
+        FORCAST_VERSION_PATH,
+      } = constants;
+      return FORCAST_PATH + FORCAST_VERSION_PATH;
+    },
+    getAPIUrl(category, path) {
+      const {
+        BASE_URL,
+        Category,
+      } = constants;
+      const route = (category === Category.GEO && this.getLocationAPI() )
+      || (category === Category.WEATHER && this.getWeatherAPI() )
+      || (category === Category.FORCAST && this.getForcastAPI() );
+      return BASE_URL + route + path;
+    },
+    async makeRequest(args = {}) {
+      const {
+        $ = this,
+        method = "get",
+        path,
+        params,
+        data,
+        category,
+      } = args;
+      const config = {
+        method,
+        url: this.getAPIUrl(category, path),
+        headers: this.getHeader(),
+        params,
+        data,
+      };
+      return axios($, config);
+    },
+    async getLocationCordinate(params) {
+      const {  Category } = constants;
+      return this.makeRequest({
+        path: "/direct",
+        params,
+        category: Category.GEO,
+      });
+    },
```

Comment:
Need to pass along `$` to `makeRequest`.
```suggestion
    async getLocationCordinate(params, $) {
      const {  Category } = constants;
      return this.makeRequest({
        path: "/direct",
        params,
        category: Category.GEO,
        $,
      });
    },
```

Reply:
Kindly check the implementation for "$".

![image](https://user-images.githubusercontent.com/68253993/167494801-62115e4b-fde2-4817-ad6a-8ac8beb8affe.png)

Kindly explain what the difference between the "this" and "$" context. I am just a little confused because passing "this" to axios worked.

Reply:
The `$` variable gives us better debugging info from axios, but it's only used in actions. Either `this` or `$` will work, so we tell `makeRequest` to use `this` if `$` isn't available. For instance if you're running a source, it has to use `this`, but if we're running an action, we can pass in `$`.

Reply:
Okay. Thanks. I understand now